cmd/oras/root/push.go

```diff
@@ -116,7 +122,7 @@ Example - Push file "hi.txt" into an OCI image layout folder 'layout-dir' with t
 				return err
 			}
 
-			if opts.manifestConfigRef != "" && opts.artifactType == "" {
+			if (opts.manifestConfigRef != "" || opts.Platform.Platform != nil) && opts.artifactType == "" {
```

Comment:
This block handles `oras push --config xxx --artifact-type xxx ...`: 

- For v1.1 artifacts, this usage invalid. 
- Otherwise best-effort try to pack v1.0 artifacts.


Since `--artifact-platform` cannot be used with `--config`, there is no need to change this code block.

Reply:
Only need to make sure: `--artifact-platform` is not used for v1.0 artifact in the below `opts.PackVersion` switching block.

Reply:
Made the change I think you requested. I do need to think about this some more though.

Reply:
> Only need to make sure: --artifact-platform is not used for v1.0 artifact in the below opts.PackVersion switching block.

This is wrong actually. Sorry for the back-and-forth change.